eZ/Publish/Core/Repository/Permission/PermissionResolver.php

```diff
@@ -62,21 +64,23 @@ class PermissionResolver implements PermissionResolverInterface
      * @param \eZ\Publish\Core\Repository\Helper\RoleDomainMapper $roleDomainMapper
      * @param \eZ\Publish\Core\Repository\Permission\LimitationService $limitationService
      * @param \eZ\Publish\SPI\Persistence\User\Handler $userHandler
-     * @param \eZ\Publish\API\Repository\Values\User\UserReference $userReference
-     * @param array $policyMap Map of system configured policies, for validation usage.
+     * @param array $policyMap
      */
     public function __construct(
         RoleDomainMapper $roleDomainMapper,
         LimitationService $limitationService,
         UserHandler $userHandler,
-        APIUserReference $userReference,
-        array $policyMap = []
+        ConfigResolverInterface $configResolver,
+        array $policyMap
     ) {
         $this->roleDomainMapper = $roleDomainMapper;
         $this->limitationService = $limitationService;
         $this->userHandler = $userHandler;
-        $this->currentUserRef = $userReference;
         $this->policyMap = $policyMap;
+
+        $this->currentUserRef = new UserReference(
+            $configResolver->getParameter('anonymous_user_id')
```

Comment:
You shouldn't do it in a constructor, otherwise we'll end up with making PermissionResolver lazy again at some point (the issue occurs randomly based on container changes affecting stack trace a.k.a the order of things)